cdap-unit-test/src/test/java/co/cask/cdap/flow/stream/FlowStreamIntegrationTestRun.java

```diff
@@ -40,7 +41,29 @@ public void testStreamBatch() throws Exception {
     for (int i = 0; i < 50; i++) {
       s1.send(String.valueOf(i));
     }
+    submitAndVerifyFlowProgram(applicationManager);
+  }
+
+  @Test
+  public void testStreamFromOtherNamespaceBatch() throws Exception {
+    NamespaceId streamSpace = new NamespaceId("streamSpace");
+    createNamespace(streamSpace.toId());
+    // Deploy an app to add a stream in streamSpace
+    deployApplication(streamSpace.toId(), ConnectStreamspaceFlowApp.class);
+    
+    ApplicationManager applicationManager = deployApplication(ConnectStreamspaceFlowApp.class);
+    StreamManager s1 = getStreamManager(streamSpace.toId(), "s1");
+    StreamManager s1Default = getStreamManager("s1");
+    // Send to both stream
+    for (int i = 0; i < 50; i++) {
+      s1.send(String.valueOf(i));
+      s1Default.send(String.valueOf(i));
+    }
+
+    submitAndVerifyFlowProgram(applicationManager);
+  }
```

Comment:
This case is included I think. In line 55, 56, two streams with the same name "s1" from different namespaces are  accessed. I send value to both of these streams and get consumed by the program. Is that enough?